pkg/services/sqlstore/user.go

```diff
@@ -158,6 +197,12 @@ func (ss *SQLStore) GetUserById(ctx context.Context, query *models.GetUserByIdQu
 			return models.ErrUserNotFound
 		}
 
+		if ss.Cfg.CaseInsensitiveLogin {
+			if err := ss.userCaseInsensitiveLoginConflict(ctx, sess, user.Login, user.Email); err != nil {
+				return err
+			}
+		}
+
```

Comment:
What's the intent here exactly? This seems like it would have the effect of preventing loading these users at all, unless I'm misinterpreting things.

Reply:
The objective would be in case there’s a duplicate, both users should be blocked from logging in until one of the duplicate accounts is deleted or merged. 

This seemed like the safest way to prevent unintended cross account access, most likely they would be the same user but it's also better for the admin to make that call.